src/libsync/account.cpp

```diff
@@ -374,8 +374,11 @@ void Account::slotHandleSslErrors(QNetworkReply *reply , QList<QSslError> errors
     // Keep a ref here on our stackframe to make sure that it doesn't get deleted before
     // handleErrors returns.
     QSharedPointer<QNetworkAccessManager> qnamLock = _am;
+    QPointer<QObject> guard = reply;
 
     if (_sslErrorHandler->handleErrors(errors, reply->sslConfiguration(), &approvedCerts, sharedFromThis())) {
+        if (!guard) return;
```

Comment:
@ogoffart I meant that we should move the check to before `reply->ignoreSslErrors()`. That way we can store the approved cert and save the account. The user dialog probably went through just fine. And if we don't do it, the user will be asked about the same certificate again.